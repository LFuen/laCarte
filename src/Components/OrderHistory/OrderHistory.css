.list {
    display: flex;
    justify-content: center;
    flex-direction: column;
    align-items: center;
    margin-top: 25px;
    width: 100%;
    margin-bottom: 100px;
}

.eachOrder {
    background-color: rgb(70, 70, 70);
    border-radius: 40px;
    border: 3px solid rgb(233, 111, 111);
    color:rgb(233, 111, 111);
    text-shadow: 2px 2px black;
    text-align: center;
    box-shadow: inset 2px 2px 2px 2px black;
    width: 100%;
    margin-bottom: 20px;
    padding: 15px;
}


.order {
    margin: 5px;
}

p.order {
    font-size: 15px;
    font-weight: bold;
    color: rgb(255, 255, 255) !important;
    text-shadow: 2px 2px black;
}

button.shadow {
    margin: 10px;
}

.noOrder {
    margin-top: 50px;
}