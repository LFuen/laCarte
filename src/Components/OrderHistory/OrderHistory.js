import React, {Component} from 'react'
import api from '../../config'
import LaCarteContext from '../../context/LaCarteContext'
import {Link} from 'react-router-dom'
import './OrderHistory.css'


class OrderHistory extends Component{

    static defaultProps = {
        match: {
            params: {}
        }
    }

    static contextType = LaCarteContext

    state = {
        error: null
    }


    clickDelete = id => {

        this.setState({error: null})
        fetch(`${api.API_ENDPOINT}/api/orders/${id}`, {
            method: 'DELETE',
            headers: {
                'Authorization': `Bearer ${api.API_KEY}`,
                'content-type' : 'application/json'
            }
        })
        .then(res => {
            if(!res.ok)
                return res.json().then(e => Promise.reject(e))
        })
        .then(() => {
            this.context.deleteOrder(id)
        })
        .catch(error => {
            console.error({error})
            this.setState({error})
        })
    }

    render() {

        const {orders} = this.context
        
        if(orders.length > 0){
        return(
            <div className='list'>
                <div className='allOrders'>
                    {orders.map(order => { return (
                    <div key={order.id} className='eachOrder'>
                        <h4>{order.meal}</h4>
                        <p className='order'>Address: {order.prim_add}</p>
                        <p className='order'>Phone: {order.phone}</p>
                        <Link to={`/orders/${order.id}/${order.meal}`}><button type='button' className='shadow'>Edit</button></Link>
                        <button type='button' className='shadow' onClick={() => this.clickDelete(order.id)}>
                                Delete
                        </button>
                    </div>
                    )})}            
                </div>
            </div>
        )}else if(orders.length <= 0){
            return (<img alt='No Orders' className='noOrder' src='https://lfuen.github.io/laCarte/images/noOrder.png'/>)
        }
    }
}



export default OrderHistory