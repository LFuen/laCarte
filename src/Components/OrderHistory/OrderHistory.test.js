import React from 'react'
import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer'
import ShallowRenderer from 'react-test-renderer/shallow'
import { BrowserRouter } from 'react-router-dom';
import LaCarteContext from '../../context/LaCarteContext'
import OrderHistory from './OrderHistory'



describe(`OrderHistory Component`, () => {
    describe(`Snapshot`, () => {
        it('renders the UI as expected', () => {
            const order = renderer.create(<BrowserRouter><OrderHistory /></BrowserRouter>).toJSON()
                expect(order).toMatchSnapshot()
        })
    })

    describe(`Smoke test`, () => {
        it(`Renders without crashing`, () => {
            
            const div = document.createElement('div')
            const renderer = new ShallowRenderer()

            const value = {
                meals: 'string',
                chefs: 'string',
                cuisines: 'string',
                orders: 'string',
                addOrder: 'string',
                updateOrder: 'string',
                deleteOrder: 'string',
            };

            renderer.render(
                <BrowserRouter>
                    <LaCarteContext.Provider value={value}>
                        <OrderHistory />
                    </LaCarteContext.Provider>
                </BrowserRouter>,
                div)
            ReactDOM.unmountComponentAtNode(div)
        })
    })
})