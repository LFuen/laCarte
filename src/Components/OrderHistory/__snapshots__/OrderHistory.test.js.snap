// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OrderHistory Component Snapshot renders the UI as expected 1`] = `
<img
  alt="No Orders"
  className="noOrder"
  src="https://lfuen.github.io/laCarte/images/noOrder.png"
/>
`;
